zkevm-circuits/src/rlp_circuit.rs

```diff
@@ -0,0 +1,1705 @@
+                        meta.query_advice(rlp_table.tag_rindex, Rotation::next()),
+                        meta.query_advice(tag_length, Rotation::next()),
+                    );
+                }
+            );
+
+            cb.gate(meta.query_fixed(q_usable, Rotation::cur()))
+        });
+
+        meta.create_gate("DataType::TxSign (unsigned transaction)", |meta| {
+            let mut cb = BaseConstraintBuilder::default();
+
+            let (_, tindex_eq) = tag_index_cmp_1.expr(meta, None);
+
+            //////////////////////////////////////////////////////////////////////////////////////
+            ////////////////////////////////// RlpTxTag::ChainID /////////////////////////////////
+            //////////////////////////////////////////////////////////////////////////////////////
+
+            // if tag_index == 1
+            cb.condition(is_chainid(meta) * tindex_eq, |cb| {
+                // checks for RlpTxTag::Zero on the next row.
+                cb.require_equal(
+                    "next tag is Zero => tag_index::next == 1",
+                    meta.query_advice(rlp_table.tag_rindex, Rotation::next()),
+                    1.expr(),
+                );
+                cb.require_equal(
+                    "next tag is Zero => value::next == 128",
+                    meta.query_advice(byte_value, Rotation::next()),
+                    128.expr(),
+                );
+                cb.require_equal(
+                    "next tag is Zero => value_acc::next == 128",
+                    meta.query_advice(rlp_table.value_acc, Rotation::next()),
+                    0.expr(),
+                );
+
+                // checks for RlpTxTag::Zero on the next-to-next row.
+                cb.require_equal(
+                    "tag::Rotation(2) == RlpTxTag::Zero",
+                    meta.query_advice(rlp_table.tag, Rotation(2)),
+                    RlpTxTag::Zero.expr(),
+                );
+                cb.require_equal(
+                    "tag_index::Rotation(2) == tag_length::Rotation(2)",
+                    meta.query_advice(rlp_table.tag_rindex, Rotation(2)),
+                    meta.query_advice(tag_length, Rotation(2)),
+                );
+                cb.require_equal(
+                    "next-to-next tag is Zero => tag_index::Rotation(2) == 1",
+                    meta.query_advice(rlp_table.tag_rindex, Rotation(2)),
+                    1.expr(),
+                );
+                cb.require_equal(
+                    "next-to-next tag is Zero => value::Rotation(2) == 128",
+                    meta.query_advice(byte_value, Rotation(2)),
+                    128.expr(),
+                );
+                cb.require_equal(
+                    "next-to-next tag is Zero => value_acc::Rotation(2) == 128",
+                    meta.query_advice(rlp_table.value_acc, Rotation(2)),
+                    0.expr(),
+                );
+
```

Comment:
We need to require that `rindex = 1` as it has reached the last byte of an RLP instance.